src/sage/schemes/curves/constructor.py

```diff
@@ -376,5 +385,12 @@ def Curve(F, A=None):
             return ProjectivePlaneCurve_field(A, F)
         return ProjectivePlaneCurve(A, F)
 
+    elif isinstance(A, WeightedProjectiveSpace_ring):
+        # currently, we only support curves in a weighted projective plane
+        if n != 2:
+            raise NotImplementedError("ambient space has to be a weighted projective plane")
+        # currently, we do not perform checks on weighted projective curves
```

Comment:
Are there any checks that a user would reasonably expect should be performed? If so, document that these checks aren't done (and ideally why they aren't) in the docstring.